### PR TITLE
Make some modules buildable for iOS Framework.

### PR DESCRIPTION
**Pull request by kikuchy**
## Issue
- A part of #136

## Overview (Required)
- Now some modules below can build for iOS Frameworks! 📱 
    - `:data:api`
    - `:data:api-impl`
    - `:data:repository`
    - `:data:firestore`
    - `:data:db`
    - `:ext:log`
- How can we implement alternative of `:data:db-room` and `:data:firestore-impl` ...? 🙄 

## Links
- https://qiita.com/nosix/items/50166b1664e6946aafe9

### CODE REVIEW

File: data/api-impl/src/commonMain/kotlin/io/github/droidkaigi/confsched2019/data/api/ApiEndpoint.kt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package io.github.droidkaigi.confsched2019.data.api
+
+internal expect fun apiEndpoint(): String
```
Comment (kikuchy) [inline data/api-impl/src/commonMain/kotlin/io/github/droidkaigi/confsched2019/data/api/ApiEndpoint.kt]: Thanks for https://github.com/DroidKaigi/conference-app-2019/pull/246#issuecomment-452247536

**Comment by takahirom** [comment]
Probably task :data:api-impl:compileDebugUnitTestKotlinAndroid become error 👀

**Comment by takahirom** [comment]
> How can we implement alternative of :data:db-room and :data:firestore-impl.

I think we can implement DroidKaigi by using only API.
But If you want that, we should migrate to SQDelite 😭
And we should implement wrapper for firestore or Firestore.kt

File: data/api-impl/src/iosMain/kotlin/io/github/droidkaigi/confsched2019/data/api/ApiComponent.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.ios.Ios
+import io.ktor.client.features.UserAgent
+import platform.Foundation.NSBundle
+
+// TODO: Replace with DI tools.
+internal fun generateHttpClient(): HttpClient {
+    val version = NSBundle.mainBundle.objectForInfoDictionaryKey("CFBundleShortVersionString") as String
+    return HttpClient(Ios) {
+        install(UserAgent) {
```
Comment (jmatsu) [inline data/api-impl/src/iosMain/kotlin/io/github/droidkaigi/confsched2019/data/api/ApiComponent.kt]: I didn't know this. 😮 🆒

**Comment by jmatsu** [comment]
> How can we implement alternative of :data:db-room and :data:firestore-impl ...? 

As for `:data:db-room`, in-memory dictionary implementation might be enough for now? 🤔

**Comment by takahirom** [comment]
@kikuchy Thank you!! 😭 
For reducing your burden, what is the next step? If you can please open issue 🙏

**Comment by kikuchy** [comment]
@takahirom Thank you for your concern!  
Now I open next step issue https://github.com/DroidKaigi/conference-app-2019/issues/348 .

**Pull request merged**